fastar/rules.py

```diff
@@ -359,22 +342,31 @@ def conv_general_dilated_dependency_rule(
   pad_config = [(0, 0, 0)] * 2 + [
     (lo, hi, dil - 1) for (lo, hi), dil in zip(padding, lhs_dilation)]
   padded_lhs = lax.pad_p.abstract_eval(lhs, pad_val, padding_config=pad_config)
-  ((outstart, _),), (outcount,), out_transpose = transpose_dependency_rule(
+  ((outstart, _),), (outcount,), _ = transpose_dependency_rule(
     outstart, outcount, None, np.argsort(out_spec))
-  ((padded_l_start, _), (r_start, r_shape)), (padded_lhs_count, rhs_count), conv = conv_dependency_rule(
+  ((padded_l_start, _), (r_start, r_shape)), (padded_lhs_count, rhs_count) = conv_dependency_rule(
     outstart, outcount, padded_lhs, rhs, window_strides, precision)
-  (l_box, _), (lhs_count, _), lhs_pad = pad_dependency_rule(
+  (l_box, _), (lhs_count, _), (lhs_pad_kwargs, _) = pad_dependency_rule(
     padded_l_start, padded_lhs_count, lhs, pad_val, pad_config)
   if l_box is not None:
     l_start, _ = l_box
-    (l_box,), (lhs_count,), lhs_transpose = transpose_dependency_rule(
+    (l_box,), (lhs_count,), _ = transpose_dependency_rule(
       l_start, lhs_count, None, lhs_spec)
-  (r_box,), (rhs_count,), rhs_transpose = transpose_dependency_rule(
+  (r_box,), (rhs_count,), _ = transpose_dependency_rule(
     r_start, rhs_count, None, rhs_spec)
   return ([l_box, r_box], [lhs_count, rhs_count],
-          lambda lhs_slice, rhs_slice: out_transpose(conv(
-            lhs_pad(None if lhs_slice is None else lhs_transpose(lhs_slice),
-                    np.zeros((), lhs.dtype)),
-            rhs_transpose(rhs_slice))))
+          (dict(window_strides=window_strides, dimension_numbers=dimension_numbers, precision=precision, **lhs_pad_kwargs), None))
+
+def conv_general_dilated_kernel(lhs_slice, rhs_slice, window_strides, dimension_numbers, precision, **lhs_pad_kwargs):
+  lhs_spec, rhs_spec, out_spec = dimension_numbers
+  # abstract evaluations to retrieve shapes:
```

Comment:
Is there an abstract eval here? Just wondering if this comment might be out of date.